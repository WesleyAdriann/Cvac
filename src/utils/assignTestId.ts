import { Platform } from 'react-native'

interface ITestID {
  accessible?: boolean
  testID: string
  accessibilityLabel?: string
}

type TComponentType =
  | 'View'
  | 'ScrollView'
  | 'TextInput'
  | 'Text'
  | 'SafeAreaView'
  | 'Pressable'
  | 'Modal'

export const assignTestId = (
  componentType: TComponentType,
  componentId: string
): ITestID => {
  const id = `${componentType}_${componentId}`
  return Platform.OS === 'android'
    ? {
        testID: id,
        accessible: true,
        accessibilityLabel: id
      }
    : {
        testID: id,
        accessible: false
      }
}
export const mockTestID = (
  componentType: TComponentType,
  componentId: string
): ITestID => assignTestId(componentType, componentId)
