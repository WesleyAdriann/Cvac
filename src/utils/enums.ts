export enum ECalendarsName {
  'kid' = 'Crian√ßa',
  'elder' = 'Idoso',
  'adult' = 'Adulto',
  'teen' = 'Adolescente'
}
