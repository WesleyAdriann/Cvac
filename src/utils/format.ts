export const formatDate = (value: string) => {
  const clearDate = value.replace(/\D/g, '')

  if (clearDate.length === 8) return value

  const day = clearDate.slice(0, 2)
  const month = clearDate.slice(2, 4)
  const year = clearDate.slice(4, 8)

  if (!month) return day
  if (!year) return `${day}/${month}`
  return `${day}/${month}/${year}`
}
