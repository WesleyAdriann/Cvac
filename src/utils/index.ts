export * from './assignTestId'
export * from './withProviders'
export * from './format'
export * from './logs'
