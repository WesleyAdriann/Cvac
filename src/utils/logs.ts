export const log = (TAG: string, ...args: any[]) => {
  if (__DEV__) console.log(`[${TAG}] `, ...args)
}
