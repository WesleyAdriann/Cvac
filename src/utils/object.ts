export const isEmpty = (obj: object | null) => {
  if (!obj) return true
  // eslint-disable-next-line no-unreachable-loop
  for (const key in obj) return false
  return true
}
