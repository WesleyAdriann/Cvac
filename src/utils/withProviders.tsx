import React from 'react'
import { Provider as PaperProvider } from 'react-native-paper'
import { ThemeProvider } from 'styled-components'

import { theme, paperTheme } from '../tokens'

export const TestsProviders: React.FC = ({ children }) => (
  <PaperProvider theme={paperTheme} >
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </PaperProvider>
)
