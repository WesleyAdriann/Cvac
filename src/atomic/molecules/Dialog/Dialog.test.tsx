import React from 'react'
import { render, cleanup } from '@testing-library/react-native'

import { AllTheProviders as wrapper, mockTestID } from '~/utils'

import { Dialog, IDialog } from './index'

describe('Molecules/Dialog', () => {
  const props: IDialog = {
    testID: 'Dialog',
    visible: true,
    content: 'Title',
    onClose: jest.fn()
  }

  const elContainer = mockTestID('Modal', props.testID!).testID

  const renderComponent = () => render(<Dialog {...props} />, { wrapper })
  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  it(`should render component #${elContainer}`, async () => {
    const sut = await component.findByTestId(elContainer)
    expect(sut).toBeTruthy()
  })

  it('should render snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
