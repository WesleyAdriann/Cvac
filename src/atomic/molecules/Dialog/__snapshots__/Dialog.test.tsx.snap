// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Dialog should render snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    accessible={false}
    animationType="fade"
    hardwareAccelerated={false}
    testID="Modal_Dialog"
    transparent={true}
    visible={true}
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      nativeID="animatedComponent"
      onAccessibilityEscape={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityLabel="Close modal"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        importantForAccessibility="no"
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.5)",
            "flex": 1,
            "opacity": 1,
          }
        }
      />
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "marginBottom": 0,
              "marginTop": 20,
            },
            undefined,
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 4,
              "elevation": 24,
              "justifyContent": "flex-start",
              "marginHorizontal": 26,
              "marginVertical": 0,
              "opacity": 1,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 23,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 24,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 24,
                  "paddingHorizontal": 24,
                },
                Array [
                  Object {
                    "paddingTop": 24,
                  },
                  undefined,
                ],
              ]
            }
          >
            <Text
              accessible={false}
              style={
                Array [
                  Object {
                    "color": "#424242",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "color": "#424242",
                      "fontSize": 16,
                      "textAlign": "left",
                      "textDecorationColor": "black",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                    },
                  ],
                ]
              }
              testID="Text_Text"
            >
              Title
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "padding": 8,
                },
                undefined,
              ]
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "#00C9D2",
                  "borderRadius": 4,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "elevation": 0,
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "minWidth": "auto",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 4,
                    },
                  ]
                }
                testID="Pressable_Button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      Object {
                        "paddingVertical": 6,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#424242",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "letterSpacing": 1,
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          Object {
                            "marginHorizontal": 8,
                          },
                          Object {
                            "textTransform": "uppercase",
                          },
                          Object {
                            "color": "#00C9D2",
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          Object {
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          Object {
                            "color": "#424242",
                            "fontSize": 18,
                          },
                        ],
                      ]
                    }
                  >
                    <Text
                      mode="text"
                      style={
                        Array [
                          Object {
                            "color": "#424242",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "color": "#424242",
                              "fontSize": 18,
                              "fontWeight": "normal",
                            },
                          ],
                        ]
                      }
                    >
                      Ok
                    </Text>
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
